### PR TITLE
fix(text_edit): discarded change from the initial buffer

### PR DESCRIPTION
**Pull request by tbruyelle**
When multiple calls to the `apply_text_edits()` function are made, the changes made to the current buffer are discarded if the following `text_edit` targets a file that is not in the buffer list.

The problem comes from the `_switch` function, which executes `edit!` when it detects that the target is not in the buffer list. If a change has already been made to the current buffer before that, that change is discarded (definition of `edit!`).

The fix uses of a different logic: if `_switch` detects that the target is not in the buffer list, it calls `badd` before switching to the buffer that has just been added. No more `edit!`, no more discarded changes.

The added test will fail without the patch in the `_switch` function.

### CODE REVIEW

**Comment by tbruyelle** [comment]
Hey @prabirshrestha, could you please review  this PR ?
As mentioned in the body, the test can reproduce the bug, so if you want to make sure that this bug exists,  just pull and run the test, without the fix. 
Thanks in advance :pray:

File: test/lsp/utils/text_edit.vimspec [hunk 1]
```diff
@@ -2,6 +2,7 @@ function! s:set_text(lines)
     % delete _
     put =a:lines
     execute 'normal ggdd'
+    execute 'write! /tmp/xyz'
```
Comment (mattn) [inline test/lsp/utils/text_edit.vimspec:5]: Is this line needed?

Comment (tbruyelle) [reply test/lsp/utils/text_edit.vimspec:5]: That said there's maybe a better way to give a name to a buffer, without involving a disk write.

Comment (tbruyelle) [reply test/lsp/utils/text_edit.vimspec]: @mattn I have changed to `file my-file` which has the benefit of non affecting the file system.
[7fce3b8](https://github.com/prabirshrestha/vim-lsp/pull/1552/commits/7fce3b8010b029843fdd7ccf53d3f55bd537e512)[7fce3b8](https://github.com/prabirshrestha/vim-lsp/pull/1552/commits/7fce3b8010b029843fdd7ccf53d3f55bd537e512)

**Comment by tbruyelle** [comment]
Hey there, any chance to have a review for this PR plz ?

**Comment by prabirshrestha** [comment]
Merged. Thanks.

**Pull request merged**